agent/session.go

```diff
@@ -402,6 +405,9 @@ func (s *session) sendError(err error) {
 // of event loop.
 func (s *session) close() error {
 	s.closeOnce.Do(func() {
+		if s.cancel != nil {
```

Comment:
if we do away with the nil check, it needs to be well-documented in the code that this field can never be nil.